content/en/docs/concepts/overview/working-with-objects/names.md

```diff
@@ -42,6 +41,36 @@ spec:
     - containerPort: 80
 ```
 
+Below are three types of commonly used name constraints for resources.
+
+### DNS Subdomain Names
+
+Most resource types require a name that can be used as a DNS subdomain name
+as defined in [RFC 1123](https://tools.ietf.org/html/rfc1123).
+This means the name must:
+
+- contain at most 253 characters
+- contain only lower case alphanumeric characters, '-' or '.'
+- start with an alphanumeric character
+- end with an alphanumeric character
+
+### DNS Label Names
+
+There are some resource types requiring their names to conform to the DNS
+label standard as defined in [RFC 1123](https://tools.ietf.org/html/rfc1123).
+This means the name must:
+
+- contain at most 63 characters
+- contain only lower case alphanumeric characters or '-'
+- start with an alphanumeric character
+- end with an alphanumeric character
+
+### Path Segment Names
+
+Some resource types require their names to be able to be safely encoded as a
+path segment. In other words, the name may not be "." or "..". and the name may
```

Comment:
nit: Possible cleanup, remove extra `.` after `..`.
Should the characters use single or double quotes?
Some Kubernetes resource types (including Custom Resources?) require their names to be safely encoded as a path segment. In other words, the resource name may not be `.` or `..` and the name may ...

Reply:
- extra "." removed
- single quotes are often used to quote a character, double quotes are often used to quote a (sub)string, so I'm using double quotes here